packages/compose/src/hooks/use-debounce/index.js

```diff
@@ -0,0 +1,46 @@
+/**
+ * useDebounce hook
+ */
+
+/**
+ * WordPress dependencies
+ */
+import { useEffect } from '@wordpress/element';
+
+/**
+ * Use to debounce a value that is changing too frequently.
+ *
+ * Example usage:
+ *
+ * ```js
+ *	// An example using the useDebounce hook with useResizeObserver()
+ *  // to debounce a rapidly changing width value
+ *
+ *	const [ resizeListener, sizes ] = useResizeObserver();
+ *
+ *	useDebounce( ( sz ) => {
+ *
+ *		// ... do your thing here ...
+ *
+ *	}, 200, sizes.width );
+ *
+ *	return (
+ *		<div>
+ *			{resizeListener}
+ *		</div>
+ *	);
+ *	```
+ *
+ * @param {Function} callback The function to call with the debounced value.
+ * @param {number}   delay    The amount to delay time in ms.
+ * @param {*}        deps     The dependent value changing you want to debounce.
```

Comment:
I think most hooks would be expected to receive `deps` as an array of dependencies, not a singular value.

Should be a small change on our part here, I'd imagine.

Reply:
Switched to array :+1: